agenda-connectors-services/src/main/java/org/exoplatform/agendaconnector/service/ExchangeConnectorServiceImpl.java

```diff
@@ -134,5 +145,46 @@ private ExchangeService connectExchangeServer(ExchangeService exchangeService,
     exchangeService.setUrl(new URI(exchangeServerURL + ExchangeConnectorUtils.EWS_URL));
     exchangeService.getInboxRules();
     return exchangeService;
+
   }
+
+  @Override
+  public void pushEventToExchange(long identityId, EventEntity event, ZoneId userTimeZone) throws IllegalAccessException {
+    ExchangeUserSetting exchangeUserSetting = getExchangeSetting(identityId);
+
+    try (ExchangeService exchangeService = new ExchangeService(ExchangeVersion.Exchange2010_SP2)) {
+      connectExchangeServer(exchangeService, exchangeUserSetting);
+      RemoteEvent remoteEvent = agendaRemoteEventService.findRemoteEvent(event.getId(), identityId);
+      if(remoteEvent == null){
+        Appointment meeting = new Appointment(exchangeService);
+        meeting.setSubject(event.getSummary());
+        ZonedDateTime startDate = AgendaDateUtils.parseRFC3339ToZonedDateTime(event.getStart(), userTimeZone);
+        ZonedDateTime endDate = AgendaDateUtils.parseRFC3339ToZonedDateTime(event.getEnd(), userTimeZone);
+        meeting.setStart(AgendaDateUtils.toDate(startDate));
+        meeting.setEnd(AgendaDateUtils.toDate(endDate));
+        meeting.save(new FolderId(WellKnownFolderName.Calendar), SendInvitationsMode.SendToAllAndSaveCopy);
+        RemoteEvent remoteEvent1 = new RemoteEvent();
```

Comment:
```suggestion
        remoteEvent = new RemoteEvent();
```
you can use the same variable remoteEvent without creating a new one